src/masternodes/mn_checks.cpp

```diff
@@ -922,28 +922,32 @@ class CCustomTxApplyVisitor : public CCustomTxVisitor
             return Res::Err("Masternode %s is not in 'ENABLED' state", obj.mnId.ToString());
         }
 
-        if (obj.firstType == static_cast<uint8_t>(UpdateMasternodeType::None) &&
-            obj.secondType == static_cast<uint8_t>(UpdateMasternodeType::None)) {
-            return Res::Err("Missing both rewardAddress or operatorAddress arguments");
+        if (obj.updates.empty()) {
+            return Res::Err("No update arguments provided");
         }
 
-        if (obj.firstType == static_cast<uint8_t>(UpdateMasternodeType::OperatorAddress)) {
-            if (obj.operatorType != 1 && obj.operatorType != 4) {
-                return Res::Err("Operator address must be P2PKH or P2WPKH type");
-            }
-            if (mnview.GetMasternodeIdByOwner(obj.operatorAddress) || mnview.GetMasternodeIdByOperator(obj.operatorAddress)) {
-                return Res::Err("Masternode with that operator address already exists");
-            }
-            mnview.UpdateMasternode(obj.mnId, *node, obj.operatorType, obj.operatorAddress, height);
-        }
-
-        if (obj.secondType == static_cast<uint8_t>(UpdateMasternodeType::SetRewardAddress)) {
-            if (obj.rewardType != 1 && obj.rewardType != 4) {
-                return Res::Err("Reward address must be P2PKH or P2WPKH type");
+        for (const auto& item : obj.updates) {
+            LogPrintf("TYPE: %d\n", item.first);
```

Comment:
We can remove the log.

Reply:
Oops... Was trying to figure out why the updated Python test was not working with incorrect type set. Did you spot the empty else statement yet?